internal/mode/static/state/change_processor.go

```diff
@@ -151,12 +152,12 @@ func NewChangeProcessorImpl(cfg ChangeProcessorConfig) *ChangeProcessorImpl {
 			{
 				gvk:       extractGVK(&apiv1.Service{}),
 				store:     newObjectStoreMapAdapter(clusterStore.Services),
-				predicate: nil,
+				predicate: funcPredicate{stateChanged: isReferenced},
 			},
 			{
 				gvk:       extractGVK(&discoveryV1.EndpointSlice{}),
-				store:     nil,
-				predicate: nil,
+				store:     newObjectStoreMapAdapter(clusterStore.EndpointSlices),
```

Comment:
I don't think we should store EndpointSlices. EndpointSlices work a bit differently than our other resources. We resolve Services to endpoints when we [buildUpstreams](https://github.com/nginxinc/nginx-gateway-fabric/blob/512b54a49b0148c5eba9d006920c340a35de015b/internal/mode/static/state/dataplane/configuration.go#L369) by listing the EndpointSlices that match the Service name. We don't use the store to resolve Services to Endpoints. 

I suggest keeping the store as nil here and modifying the `changeTrackingUpdater` to make sure we still report a change for EndpointSlices when the predicate returns true.